winbuild/build_prepare.py

```diff
@@ -763,6 +750,8 @@ def main() -> None:
     }
 
     for k, v in DEPS.items():
+        if "dir" not in v:
+            v["dir"] = re.sub(r"\.(tar\.gz|zip)", "", v["filename"])
```

Comment:
Or with https://docs.python.org/3/library/os.path.html#os.path.basename ?
```suggestion
            v["dir"] = os.path.basename(v["filename"])
```

Reply:
`os.path.basename` is for removing the directories before the filename, which isn't the intention here. I presume you meant `os.path.splitext`. Sadly, that won't suffice either.

```pycon
>>> import os
>>> os.path.splitext("a.tar.gz")
('a.tar', '.gz')
```